pilot/pkg/config/kube/gateway/conditions.go

```diff
@@ -134,8 +134,8 @@ type condition struct {
 	// error defines an error state; the reason and message will be replaced with that of the error and
 	// the status inverted
 	error *ConfigError
-	// setOnce, if enabled, will only set the condition if it is not yet present
-	setOnce bool
+	// setOnce, if enabled, will only set the condition if it is not yet present or set to this reason
+	setOnce string
```

Comment:
This is more like a bool var

Reply:
Yeah let me rename it. Its kind of a bool+string - it only takes effect if its not `""`